src/fr/uca/jgit/model/Commit.java

```diff
@@ -112,4 +112,36 @@ void checkout() {
         this.state.restore("");
 
     }
+
+    /**
+     * Create a new branch if not exist
+     * @param branchName - name of the branch to create
+     * @throws IOException
+     */
+    public static void createBranch(String branchName) throws IOException {
+        Commit commit = new Commit();
+        commit.store();
+        String commitHash = commit.hash();
+        createBranch(branchName, commitHash);
+    }
+
+    /**
+     * Create a new branch if not exist
+     * @param branchName - name of the branch to create
+     * @param commitHash - commit hash of the branch
+     * @throws IOException
+     */
+    public static void createBranch(String branchName, String commitHash) throws IOException {
+        // Create a new file in .git/refs/heads with the given branch name
+        File branchFile = new File(".git/refs/heads/" + branchName);
```

Comment:
".jgit" à la place de ".git"
String filePath = Paths.get(".jgit", "object", this.hash()).toString(); pour le path

Reply:
Il y a un dernier commit [fix: compatibility of paths on different systems](https://github.com/JustSeenn/JGIT/pull/21/commits/b3cb9187c053daaab2dad835c4fbaffb8711f64d) qui résout ceci